src/main/java/com/sintern/security/filters/AuthorizationFilter.java

```diff
@@ -0,0 +1,66 @@
+package com.sintern.security.filters;
+
+import com.sintern.security.jwt.JwtUtil;
+import com.sintern.security.service.SUserDetailsService;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Arrays;
+
+@Component
+@RequiredArgsConstructor
```

Comment:
Is this actually working? It seems we have already defined a constructor with all arguments.